src/NuGet.Server.Core/Infrastructure/ServerPackageRepository.cs

```diff
@@ -593,16 +593,19 @@ private void UnregisterFileSystemWatcher()
 
                 _logger.Log(LogLevel.Verbose, "File system changed. File: {0} - Change: {1}", e.Name, e.ChangeType);
 
+                var changedDirectory = Path.GetFullPath(Path.GetDirectoryName(e.FullPath) ?? throw new InvalidOperationException());
+                var watchDirectory = Path.GetFullPath(_fileSystemWatcher.Path);
```

Comment:
This can be defined one as a field at the same time as initialization of `_fileSystemWatcher` itself.

Reply:
Promoted `_watchDirectory` to a private field.